src/shared/components/ScrollTable/ScrollTable.tsx

```diff
@@ -237,3 +270,48 @@ export const ScrollTableCore = <T extends object>({
         </ScrollTableCellsWrap>
     )
 }
+
+const OrderCaret: FunctionComponent<{
+    orderDirection: ScrollTableOrderDirection
+}> = ({ orderDirection }) => {
+    return <OrderCaretIcon name={'caretUp'} className={orderDirection} />
+}
+
+const ORDER_ITERATION_CYCLE = [
+    ScrollTableOrderDirection.asc,
+    ScrollTableOrderDirection.desc,
+    undefined,
+]
+export const getNextSortingParameters = (
+    currentOrderBy: string | undefined,
+    newOrderBy: string,
+    currentOrderDirection: ScrollTableOrderDirection | undefined,
+): {
+    orderBy: string | undefined
+    orderDirection: ScrollTableOrderDirection | undefined
+} => {
+    if (currentOrderBy !== newOrderBy) {
+        return {
+            orderBy: newOrderBy,
+            orderDirection: ORDER_ITERATION_CYCLE[0],
+        }
+    }
+    const currentDirectionCycleIndex = ORDER_ITERATION_CYCLE.findIndex(
+        (el) => el === currentOrderDirection,
+    )
+    const newDirectionCycleIndex =
+        currentDirectionCycleIndex + 1 == ORDER_ITERATION_CYCLE.length
+            ? 0
+            : currentDirectionCycleIndex + 1
+    const newDirection = ORDER_ITERATION_CYCLE[newDirectionCycleIndex]
+    if (!newDirection) {
+        return {
+            orderBy: undefined,
+            orderDirection: undefined,
+        }
+    }
+    return {
+        orderBy: newOrderBy,
+        orderDirection: newDirection,
+    }
+}
```

Comment:
```ts
export const getNextSortingParameters = (
    currentOrderBy: string | undefined,
    newOrderBy: string,
    currentOrderDirection: ScrollTableOrderDirection | undefined,
): {
    orderBy: string | undefined
    orderDirection: ScrollTableOrderDirection | undefined
} => {
    const orderDirection =
        !currentOrderDirection || currentOrderBy !== newOrderBy
            ? ScrollTableOrderDirection.asc
            : currentOrderDirection === ScrollTableOrderDirection.desc
            ? void 0
            : ScrollTableOrderDirection.desc

    return {
        orderBy: orderDirection ? newOrderBy : void 0,
        orderDirection,
    }
}

```

Reply:
TBH for me that's less readable

Reply:
All I'm saying is… let's find a way without `findIndex` man. Have a map

```ts
const map = {
    asc: 'desc',
    desc: undefined,
    '': 'asc'
}[current || '']
```

That should do it.